github/security_advisories_test.go

```diff
@@ -315,3 +317,379 @@ func TestSecurityAdvisoriesService_ListRepositorySecurityAdvisories(t *testing.T
+	want := []*GlobalSecurityAdvisory{
+		{
+			ID: Int64(1),
+			SecurityAdvisory: SecurityAdvisory{
+				GHSAID:      String("GHSA-xoxo-1234-xoxo"),
+				CVEID:       String("CVE-xoxo-1234"),
+				URL:         String("https://api.github.com/advisories/GHSA-xoxo-1234-xoxo"),
+				HTMLURL:     String("https://github.com/advisories/GHSA-xoxo-1234-xoxo"),
+				Severity:    String("high"),
+				Summary:     String("Heartbleed security advisory"),
+				Description: String("This bug allows an attacker to read portions of the affected server’s memory, potentially disclosing sensitive information."),
+				Identifiers: []*AdvisoryIdentifier{
+					{
+						Type:  String("GHSA"),
+						Value: String("GHSA-xoxo-1234-xoxo"),
+					},
+					{
+						Type:  String("CVE"),
+						Value: String("CVE-xoxo-1234"),
+					},
+				},
+				PublishedAt: &date,
+				UpdatedAt:   &date,
+				WithdrawnAt: nil,
+				CVSS: &AdvisoryCVSS{
+					VectorString: String("CVSS:3.1/AV:N/AC:H/PR:H/UI:R/S:C/C:H/I:H/A:H"),
+					Score:        Float64(7.6),
+				},
+				CWEs: []*AdvisoryCWEs{
+					{
+						CWEID: String("CWE-400"),
+						Name:  String("Uncontrolled Resource Consumption"),
+					},
+				},
+			},
+			References: []string{"https://nvd.nist.gov/vuln/detail/CVE-xoxo-1234"},
+			Vulnerabilities: []*Vulnerabilities{
+				{
+					Package: &VulnerabilityPackage{
+						Ecosystem: String("npm"),
+						Name:      String("a-package"),
+					},
+					FirstPatchedVersion:    String("1.0.3"),
+					VulnerableVersionRange: String("<=1.0.2"),
+					VulnerableFunctions:    []string{"a_function"},
+				},
+			},
+			RepositoryAdvisoryURL: String("https://api.github.com/repos/project/a-package/security-advisories/GHSA-xoxo-1234-xoxo"),
+			Type:                  String("reviewed"),
+			SourceCodeLocation:    String("https://github.com/project/a-package"),
+			GitHubReviewedAt:      &date,
+			NVDPublishedAt:        &date,
+			Credits: []*Credits{
+				{
+					User: &User{
+						Login:             String("user"),
+						ID:                Int64(1),
+						NodeID:            String("12="),
+						AvatarURL:         String("a"),
+						GravatarID:        String(""),
+						URL:               String("a"),
+						HTMLURL:           String("b"),
+						FollowersURL:      String("b"),
+						FollowingURL:      String("c"),
+						GistsURL:          String("d"),
+						StarredURL:        String("e"),
+						SubscriptionsURL:  String("f"),
+						OrganizationsURL:  String("g"),
+						ReposURL:          String("h"),
+						EventsURL:         String("i"),
+						ReceivedEventsURL: String("j"),
+						Type:              String("User"),
+						SiteAdmin:         Bool(false),
+					},
+					Type: String("analyst"),
+				},
+			},
+		},
+	}
+
+	if !cmp.Equal(advisories, want) {
+		t.Errorf("SecurityAdvisories.ListGlobalSecurityAdvisories %+v, want %+v", advisories, want)
+	}
+
+	const methodName = "ListGlobalSecurityAdvisories"
```

Comment:
```suggestion
	const methodName = "ListGlobalSecurityAdvisories"
	testBadOptions(t, methodName, func() (err error) {
		badOpts := &ListGlobalSecurityAdvisoriesOptions{CVEID: "CVE-\n-1234"}
		_, resp, err := client.SecurityAdvisories.ListGlobalSecurityAdvisories(ctx, badOpts)
		return err
	})
```

Reply:
This will return not found and not an error, cannot use this test as is, because we do not validate the options.

Reply:
Ah, interesting.  OK, thanks.